File: MovieCatalogueV2/app/src/main/java/com/google/fdp/moviecataloguev2/services/StackWidgetService.java
package com.google.fdp.moviecataloguev2.services;

import android.content.Intent;
import android.widget.RemoteViewsService;

import com.google.fdp.moviecataloguev2.widgets.StackRemoteViewsFactory;

/**
 * Created by gama on 2019-08-17.
 * Addin Gama Bertaqwa
 * addingama@gmail.com
 */
public class StackWidgetService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new StackRemoteViewsFactory(this.getApplicationContext());
    }
}

File: MovieCatalog 3/app/src/main/java/com/alfianh/moviecatalog/service/StackWidgetService.kt
package com.alfianh.moviecatalog.service

import android.content.Intent
import android.widget.RemoteViewsService
import com.alfianh.moviecatalog.widgets.StackRemoteViewsFactory

class StackWidgetService : RemoteViewsService() {
  override fun onGetViewFactory(intent: Intent?): RemoteViewsFactory = StackRemoteViewsFactory(
      applicationContext)

}